### PR TITLE
fix(ui5-menu): adjust loading sub-menu

### PR DESCRIPTION
**Pull request by unazko**
Fixes: #7391

### CODE REVIEW

File: packages/main/src/Menu.ts [hunk 1]
```diff
@@ -358,11 +359,13 @@ class Menu extends UI5Element {
 	}
 
 	get isSubMenuOpened() {
-		return this._parentMenuItem && this._popover?.isOpen();
+		return this.items.some(menuItem => menuItem.subMenuOpened);
 	}
 
 	get menuHeaderTextPhone() {
-		return this._parentMenuItem ? this._parentMenuItem.text : this.headerText;
+		const menu = this as Menu;
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:366]: This cast can be removed. This is a menu

File: packages/main/src/Menu.ts [hunk 2]
```diff
@@ -410,11 +412,14 @@ class Menu extends UI5Element {
 	 * @public
 	 */
 	async showAt(opener: HTMLElement): Promise<void> {
+		const menu = this as Menu;
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:415]: same as above

File: packages/main/src/Menu.ts [hunk 3]
```diff
@@ -442,12 +447,24 @@ class Menu extends UI5Element {
 	}
 
 	_navigateBack() {
-		this._closeItemSubMenu(this._parentMenuItem as MenuItem, true);
+		const menu = this as Menu;
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:450]: same as above

File: packages/main/src/Menu.ts [hunk 4]
```diff
@@ -442,12 +447,24 @@ class Menu extends UI5Element {
...
+		const parentMenuItem = this._getParentMenuItem(menu);
+		if (parentMenuItem) {
+			this._closeItemSubMenu(parentMenuItem, true);
+		}
 	}
 
 	_closeAll() {
-		const mainMenu = this._findMainMenu(this);
-		mainMenu?.close();
+		let menu = this as Menu,
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:458]: same as above

File: packages/main/src/Menu.ts [hunk 5]
```diff
@@ -461,22 +478,32 @@ class Menu extends UI5Element {
 	}
 
 	_createSubMenu(item: MenuItem, opener: HTMLElement) {
-		if (item._subMenu) {
-			return;
+		let subMenu = item._subMenu;
+		let subMenuRef = this._getSubmenuReference(item);
+
+		if ((!subMenu && !this._isSubMenu) || (!subMenuRef && this._isSubMenu)) {
+			const ctor = this.constructor as typeof Menu;
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:485]: const ctor = this.constructor as typeof Menu;
subMenu = document.createElement(ctor.getMetadata().getTag()) as Menu;
can be replaced with
subMenu = document.createElement(this.tagName.toLowerCase()) as Menu;

Or this.tagName.toLowerCase() can be extracted in variable, as I see it used in few times in the method.

File: packages/main/src/Menu.ts [hunk 6]
```diff
@@ -584,8 +617,10 @@ class Menu extends UI5Element {
 	}
 
 	_busyMouseOver() {
-		if (this._parentMenuItem) {
-			this._parentMenuItem._preventSubMenuClose = true;
+		const menu = this as Menu;
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:620]: same as above

**Review by tsanislavgatev — CHANGES_REQUESTED** [review]

File: packages/main/src/Menu.ts (hunk 1 above)
Comment (unazko) [reply packages/main/src/Menu.ts:366]: done

File: packages/main/src/Menu.ts (hunk 2 above)
Comment (unazko) [reply packages/main/src/Menu.ts:415]: done

File: packages/main/src/Menu.ts (hunk 5 above)
Comment (unazko) [reply packages/main/src/Menu.ts:485]: done

File: packages/main/src/Menu.ts (hunk 6 above)
Comment (unazko) [reply packages/main/src/Menu.ts:620]: done

File: packages/main/src/Menu.ts (hunk 4 above)
Comment (unazko) [reply packages/main/src/Menu.ts:458]: We need this variable for the loop.

File: packages/main/src/Menu.ts (hunk 3 above)
Comment (unazko) [reply packages/main/src/Menu.ts:450]: done

**Review by tsanislavgatev — APPROVED** [review]

**Review by ilhan007 — APPROVED** [review]

**Pull request merged**